general/management/commands/prune_database.py

```diff
@@ -130,17 +158,88 @@ def delete_downloaders(self, numkeep):
         random.shuffle(userids)
         # Keep `numkeep` users, and delete the rest
         randusers = sorted(userids[numkeep:])
-        ch = [c for c in chunks(randusers, 100000)]
+        ch = [c for c in chunks(randusers, 1000)]
         tot = len(ch)
         for i, c in enumerate(ch, 1):
             console_logger.info(f' {i}/{tot}')
             self.delete_some_users(c)
 
-    def handle(self,  *args, **options):
+    def anonymise_database(self):
+        users_to_update = []
+        for user in User.objects.all():
+            user.email = str(user.id) + '@freesound.org'
+            user.first_name = ''
+            user.last_name = ''
+            # this password is 'freesound'
+            user.password = 'pbkdf2_sha256$36000$PJRTmkaiwSEC$a8+HUj33133PZX7ToOuypT/CfLKNwMeJMXqBJ4QbQPg='
+            user.is_staff = False
+            user.is_superuser = False
+            users_to_update.append(user)
+        #User.objects.bulk_update(users_to_update, ['email', 'first_name', 'last_name', 'password', 'is_staff', 'is_superuser'])
+
+        MessageBody.objects.all().update(body='(message body) Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.')
+        TicketComment.objects.all().update(text='(ticket comment) Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.')
+
+        GdprAcceptance.objects.all().update(date_accepted=datetime.datetime.now())
+        
+        Profile.objects.all().update(
+            is_whitelisted=False,
+            not_shown_in_online_users_list=False,
+            last_stream_email_sent=None,
+            last_attempt_of_sending_stream_email=None,
+            is_adult=False,
+            last_donation_email_sent=None,
+            donations_reminder_email_sent=False,
+        )
+
+        # Bookmarks? Possibly set created to now()
+        # Messages (link between 2 users)
+
+        User.objects.all().update(last_login=datetime.datetime.now(), date_joined=datetime.datetime.now())
```

Comment:
I added this because I didn't want to leak login/creation times. Perhaps there are some other date fields which we could include in this anonymisation too